src/scaffoldmaker/meshtypes/meshtype_3d_stellate1.py

```diff
@@ -384,16 +490,65 @@ def refineMesh(cls, meshrefinement, options):
 
 
-def createArm(halfArmArcAngleRadians, elementLengths, armAngle, armAngleConst, elementsCount, dipMultiplier, armCount, armIndex):
+    @classmethod
+    def defineFaceAnnotations(cls, region, options, annotationGroups):
+        """
+        Add point annotation groups from the 1D mesh.
+        :param region: Zinc region containing model.
+        :param options: Dict containing options. See getDefaultOptions().
+        :param annotationGroups: List of annotation groups for top-level elements.
+        New point annotation groups are appended to this list.
+        """
+        # create  groups
+        fm = region.getFieldmodule()
+        armCount = len(options['Numbers of elements along arms'])
+        stellateGroup = getAnnotationGroupForTerm(annotationGroups, get_stellate_term("cervicothoracic ganglion"))
+        armNames, faceNames = getUnlabelledStellateArmNames(armCount)
+        arm1Group = getAnnotationGroupForTerm(annotationGroups, armNames[0])
+        arm2Group = getAnnotationGroupForTerm(annotationGroups, armNames[1])
+        arm3Group = getAnnotationGroupForTerm(annotationGroups, armNames[2])
+
+        mesh2d = fm.findMeshByDimension(2)
+        is_exterior = fm.createFieldIsExterior()
+        is_exterior_face_xi2_0 = fm.createFieldAnd(is_exterior, fm.createFieldIsOnFace(Element.FACE_TYPE_XI2_0))
+        is_exterior_face_xi2_1 = fm.createFieldAnd(is_exterior, fm.createFieldIsOnFace(Element.FACE_TYPE_XI2_1))
+
+        is_arm1 = arm1Group.getFieldElementGroup(mesh2d)
+        is_arm2 = arm2Group.getFieldElementGroup(mesh2d)
+        is_arm3 = arm3Group.getFieldElementGroup(mesh2d)
+        is_left_face = fm.createFieldOr(
+            fm.createFieldAnd(is_arm2, is_exterior_face_xi2_1),
+            fm.createFieldAnd(is_arm3, is_exterior_face_xi2_0))
+        leftStellateGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, faceNames[1])
+        leftStellateGroup.getMeshGroup(mesh2d).addElementsConditional(is_left_face)
+
+        is_top_face = fm.createFieldOr(
+            fm.createFieldAnd(is_arm1, is_exterior_face_xi2_1),
+            fm.createFieldAnd(is_arm2, is_exterior_face_xi2_0))
+        topStellateGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, faceNames[0])
+        topStellateGroup.getMeshGroup(mesh2d).addElementsConditional(is_top_face)
+
+        is_bottom_face = fm.createFieldOr(
+            fm.createFieldAnd(is_arm1, is_exterior_face_xi2_0),
+            fm.createFieldAnd(is_arm3, is_exterior_face_xi2_1))
+        bottomStellateGroup = findOrCreateAnnotationGroupForTerm(annotationGroups, region, faceNames[-1])
+        bottomStellateGroup.getMeshGroup(mesh2d).addElementsConditional(is_bottom_face)
+
+def getUnlabelledStellateArmNames(armCount):
+    armNames = [("stellate arm %d"%(d), None) for d in range(1,armCount+1)]
+    faceNames = [("stellate face %d-%d"%(d, d+1), None) for d in range(1,armCount+1)]
+    faceNames[-1] = ("stellate face %d-1"%(armCount), None)
```

Comment:
I suggest not using d for the indexes as it's confusing with the %d. Lines 2 and 3 can be merged using modulo operator %:
```
    faceNames = [("stellate face %d-%d"%(i, (i % armCount) + 1), None) for i in range(1, armCount + 1)]
```
(I haven't seen that way to make strings before... dinky!)
Nb. I prefer to see a space after a comma, and spaces around operators esp. + and - in expressions.